### PR TITLE
Link user and grader names in Submission Edit page

### PR DESCRIPTION
**Pull request by qreoct**
Add a link to user & grader profiles in Submission Edit page

Solves #4612

### CODE REVIEW

File: app/views/course/assessment/submission/submissions/_submission.json.jbuilder [hunk 1]
```diff
@@ -32,7 +35,10 @@ json.submission do
     # Display the published time first, else show the graded time if available.
     # For showing timestamps from delayed grade publication.
     json.gradedAt submission.published_at&.iso8601 || submission.graded_at&.iso8601
-    json.grader display_user(submission.publisher) if apparent_workflow_state == 'published'
+    json.grader do
```
[the remark this answers is not in the document]
Comment (qreoct) [reply app/views/course/assessment/submission/submissions/_submission.json.jbuilder]: Okay done!

File: client/app/bundles/course/assessment/submission/containers/GradingPanel.jsx [hunk 2]
```diff
@@ -66,6 +66,22 @@ class VisibleGradingPanel extends Component {
     }
   }
 
+  renderSubmitter() {
```
[the remark this answers is not in the document]
Comment (ekowidianto) [reply client/app/bundles/course/assessment/submission/containers/GradingPanel.jsx:69]: And dont forget to run yarn lint as well

[the remark this answers is not in the document]
Comment (qreoct) [reply client/app/bundles/course/assessment/submission/containers/GradingPanel.jsx]: Got it!

**Comment by ekowidianto** [comment]
LGTM Thanks!

**Comment by ekowidianto** [comment]
@qreoct Please help to fix the tests Thanks!

**Comment by qreoct** [comment]
Done. The issue was caused by backend not sending the `grader` object when submission is not in 'published' state, but frontend was trying to render it. I've also added a check to ensure that the 'System' user cannot be linked to in the frontend.

File: client/app/bundles/course/assessment/submission/containers/GradingPanel.jsx [hunk 3]
```diff
@@ -42,6 +42,21 @@ class VisibleGradingPanel extends Component {
...
+  static renderCourseUserLink(courseUser) {
+    const courseId = getCourseId();
+    if (courseUser && courseUser.id) {
+      return (
+        <a href={getCourseUserURL(courseId, courseUser.id)}>
+          {courseUser.name}
+        </a>
+      );
+    }
+    if (courseUser) {
```
[the remark this answers is not in the document]
Comment (qreoct) [reply client/app/bundles/course/assessment/submission/containers/GradingPanel.jsx]: Done!

**Pull request merged**